bench/workgen/runner/microbenchmark_tiered_base.py

```diff
@@ -0,0 +1,116 @@
+        self.storage_source = extension
+        self.context = Context()
+        tiered_storage_config = ",tiered_storage=(bucket=./bucket,bucket_prefix=pfx-,local_retention=0,name=dir_store)"
+        tiered_storage_config += ",extensions=(./ext/storage_sources/dir_store/libwiredtiger_dir_store.so=(early_load=true))"
+        conn_config = ",cache_size=2G,log=(enabled,file_max=10MB),statistics=(fast),statistics_log=(json,wait=1)"  # explicitly added
+        conn_config += tiered_storage_config
+        self.conn = self.context.wiredtiger_open("create," + conn_config, self.create_bucket)
+        self.session = self.conn.open_session(None)
+        wtperf_table_config = "key_format=S,value_format=S," +\
+            "exclusive=true,allocation_size=4kb," +\
+            "internal_page_max=64kb,leaf_page_max=4kb,split_pct=100,"
+        self.table_config = "type=file"
+        self.tables = []
+        self.table_count = 4
+        for i in range(0, self.table_count):
+            tname = "table:test" + str(i)
+            table = Table(tname)
+            self.session.create(tname,wtperf_table_config + self.table_config)
+            table.options.key_size = 20
+            table.options.value_size = 100
+            self.tables.append(table)
+
+        ops = Operation(Operation.OP_SEARCH, self.tables[0])
+        ops = op_multi_table(ops, self.tables, False)
+        self.read_thread = Thread(ops)
+
+        ops = Operation(Operation.OP_UPDATE, self.tables[0])
+        ops = op_multi_table(ops, self.tables, False)
+        self.update_thread = Thread(ops)
+
+        ops = Operation(Operation.OP_INSERT, self.tables[0])
+        ops = op_multi_table(ops, self.tables, False)
+        self.insert_thread = Thread(ops)
+
+    def populate(self):
+        print ("Populate tables start")
+        populate_threads = 5
+        icount = 50000
+        # There are multiple tables to be filled during populate,
+        # the icount is split between them all.
+        pop_ops = Operation(Operation.OP_INSERT, self.tables[0])
+        pop_ops = op_multi_table(pop_ops, self.tables)
+        nops_per_thread = icount // (populate_threads * self.table_count)
+        pop_ops = op_group_transaction(pop_ops, 100, "")
+        pop_thread = Thread(pop_ops * nops_per_thread)
+        pop_workload = Workload(self.context, populate_threads * pop_thread)
+        ret = pop_workload.run(self.conn)
+        assert ret == 0, ret
+        print ("Populate tables end")
+
+    def create_bucket(self, home):
+        bucket_path = home + "/" + "bucket"
+        if not os.path.isdir(bucket_path):
+            os.mkdir(bucket_path)
+
+    def set_checkpoint_thread(self, ckpt_thread):
+        self.checkpoint_thread = ckpt_thread
+
+    def print_stats(self):
+        stat_cursor = self.session.open_cursor('statistics:')
+        flush_count = stat_cursor[stat.conn.flush_tier][2]
+        stat_cursor.close()
+        assert flush_count > 2, flush_count
```

Comment:
It would be good to have a comment here.  Something like,
```
With the timing for the test run and the frequency of flush_tier, we would expect 3 or more flush_tier calls.
We'll want that many to be able examine the impact of flush_tier on response times.
If we don't see that many, it may be that individual flush_tier calls are taking a very long time.
```

Reply:
Fixed